src/modules/dashboard/components/Dialogs/EscrowFundsDialog/useEscrowFundsDialog.tsx

```diff
@@ -0,0 +1,211 @@
+          ? `${formatMessage(MSG.createDomain)} ${label}`
+          : `${formatMessage(MSG.createDomain)} ${formatMessage({
+              id: 'domain.root',
+            })}/${label}`;
+
+      if (!option) {
+        displayLabel = `${formatMessage({ id: 'domain.root' })}`;
+      }
+      return <div className={styles.motionActiveItem}>{displayLabel}</div>;
+    },
+    [formatMessage],
+  );
+
+  const filterDomains = useCallback((optionDomain) => {
+    const optionDomainId = parseInt(optionDomain.value, 10);
+
+    return optionDomainId !== COLONY_TOTAL_BALANCE_DOMAIN_ID;
+  }, []);
+
+  // @TODO replace the mock with real data
+  const balanceOptions = useMemo(
+    () =>
+      tokens.map((token, index) => ({
+        label: token.name,
+        value: token.id,
+        children: (
+          <div
+            className={classNames(styles.label, styles.option, {
+              [styles.firstOption]: index === 0,
+            })}
+          >
+            <span className={styles.icon}>
+              <TokenIcon
+                className={styles.tokenIcon}
+                token={token}
+                name={token.name || token.address}
+              />
+            </span>
+
+            <Numeral
+              unit={getTokenDecimalsWithFallback(token.decimals)}
+              value={token.balances[COLONY_TOTAL_BALANCE_DOMAIN_ID].amount}
+              className={styles.tokenNumeral}
+            />
+            <span className={styles.symbol}>{token.symbol}</span>
+          </div>
+        ),
+      })),
+    [],
+  );
+  // @TODO replace the mock with real data
+  const requiredFunds = useMemo(
+    () =>
+      requiredFundsMock.map((token) => ({
+        label: token.name,
+        value: token.id,
+        children: (
+          <FormSection appearance={{ border: 'top' }}>
+            <div className={styles.requiredFundsRow}>
+              <span>
+                {token.isPartial ? (
+                  <>
+                    <FormattedMessage
+                      {...MSG.partialFund}
+                      values={{ name: token.name }}
+                    />
+                    <span className={styles.tokenTotal}>
+                      {` (${formatMessage(MSG.total)} ${token.name} `}
+                      <Numeral
+                        className={styles.tokenNumeralTiny}
+                        unit={getTokenDecimalsWithFallback(token.decimals)}
+                        value={
+                          token.total?.[COLONY_TOTAL_BALANCE_DOMAIN_ID]
+                            .amount ?? 0
+                        }
+                      />
+                      )
+                    </span>
+                  </>
+                ) : (
+                  <FormattedMessage
+                    {...MSG.fullFund}
+                    values={{ name: token.name }}
+                  />
+                )}
+              </span>
+
+              <span className={styles.icon}>
```

Comment:
Seems that this code is repeated twice. Maybe a small component to not repeat the code?

Also please, remove the empty lines between the html elements, they are confusing as we don't usually use them.